demos/common/python/models/image_model.py

```diff
@@ -0,0 +1,67 @@
+from .model import Model
+from .utils import resize_image, resize_image_with_aspect, resize_image_letterbox, pad_image
+
+
+class ImageModel(Model):
+    RESIZE_TYPES = {
+        'default': resize_image,
+        'keep_aspect_ratio': resize_image_with_aspect,
+        'letterbox': resize_image_letterbox,
+    }
+
+    def __init__(self, ie, model_path, input_transform=None, resize_type='default'):
+        super().__init__(ie, model_path, input_transform=input_transform)
+        self.image_blob_name = self._get_image_input()
+        if self.image_blob_name:
+            self.n, self.c, self.h, self.w = self.net.input_info[self.image_blob_name].input_data.shape
+        self.image_layout = 'CHW'
+        self.resize_type = resize_type
+        self.resize = self.RESIZE_TYPES[self.resize_type]
+
+    def _get_image_input(self):
+        image_blob_name = None
+        for blob_name, blob in self.net.input_info.items():
+            if len(blob.input_data.shape) == 4:
+                if not image_blob_name:
+                    image_blob_name = blob_name
+                else:
+                    raise RuntimeError('Failed to identify the input for image: more than one 4D input layer found')
```

Comment:
what about super resolution models in future? they have 2 inputs and both images

Reply:
Done